syncd/syncd_flex_counter.cpp

```diff
@@ -119,11 +167,12 @@ void FlexCounter::removePort(
 }
 
 void FlexCounter::removeQueue(
-        _In_ sai_object_id_t queueVid)
+        _In_ sai_object_id_t queueVid,
+        _In_ uint32_t pollInterval)
 {
     SWSS_LOG_ENTER();
 
-    FlexCounter &fc = getInstance();
+    FlexCounter &fc = getInstance(pollInterval);
```

Comment:
>getInstance [](start = 22, length = 11)

if FlexCounter(pollInterval) doesn't exist, you will create one, and find nothing. The memory is never collected timely.

Reply:
Let me explain the design notes. 
Once orchagent has finished the port or queue initialization it sets the sai id with poll interval and list of stat counters  to Flex counter DB (for each port and queue). Syncd handles the events, and add the counter to appropriate Flex counter instance. We have a few FC instances for each poll interval because stat counters and PFC WD counters are polling with different intervals.  
So, FC instances are created for each poll inverval and consist own lists with ports, queues and counters to update.

Reply:
Thanks for the information. However, nothing is related to my comment above.

Reply:
@qiluo-msft 
Sorry, but I don't understand your comment.  
We have more than 500 items with the same poll interval and at least more of them will be added to the instance for sure. The instance will never live as empty. I don't see any reason to modify the code. 
Please explain more detailed.

Reply:
Let's assume in this use case:
1. You have one instance for 100 ms, and one instance for 200 ms.
2. You call getInstance(1), then it will create a new instance for 1, and add it to g_flex_counters_map. This new instance is never released.

Reply:
The g_flex_counters_map contains a shared pointers to a flex counter instances. 
The instances will be released when syncd process will finishing work. 
Stop syncd -> destroy g_flex_counters_map  -> destroy Flex counter instances.

Reply:
This is too late. It is strange that a 'get*' function will accumulate memory garbage.